src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -328,6 +334,51 @@ protected ulong ClearRegister
             set {  *(ulong*)(_registerViewPointer->GPCLR) = value; }
         }
 
+        /// <summary>
+        /// Returns the pperipheral base address of the raspberry pi based on the ranges set within the device tree.
+        /// The return value is 32bit even on 64 bit operating systems (debian / ubuntu tested) but may change in the future
+        /// 
+        /// Based on file at https://github.com/RPi-Distro/raspi-gpio/blob/master/raspi-gpio.c
```

Comment:
I'm not an expert on this, but looks like this License is not exactly MIT so I would wait for @RichardLander to confirm if we can use this as a reference or not.

Reply:
@joperezr @richlander 

I haven’t copied the code directly the reference to the code is more to indicate that we are determining the base address in a way that is consistent with the way that raspi-bipolar does it.

Reply:
I think is fine as long as we do whatever this mentions: https://github.com/RPi-Distro/raspi-gpio/blob/master/LICENSE

You might need to edit https://github.com/dotnet/iot/blob/master/THIRD-PARTY-NOTICES.TXT (use similar file in corefx for reference)